src/stock-adjustment-mobile/components/block-list.component.jsx

```diff
@@ -15,21 +15,29 @@
 
 import React from 'react';
 
-const BlockList = ({data, dataToDisplay, headerToDisplay, onClickAction}) => {
+const BlockList = ({data, dataToDisplay, headersToDisplay, onClickAction}) => {
     return (
         <div className="react-block-list-container" style={{width: "100%"}}>
             <div className="react-block-list-content">
                 {data.map((row, i) => {
+                    let headerText = "";
+                    let headersCounter = 0;
                     return (
-                        <div className="react-block-list-container-row" onClick={() => onClickAction(row, i)} style={{height: "160px", width: "100%",  borderBottom: "1px solid #dbdada", borderTop: "1px solid #dbdada"}}>    
+                        <div className="react-block-list-container-row" onClick={() => onClickAction(row, i)} style={{height: "120px", width: "100%",  borderBottom: "1px solid #dbdada", borderTop: "1px solid #dbdada"}}>    
                             {dataToDisplay.map((display, index) => {
                                 if (row.hasOwnProperty(display.key)) {
-                                    if (headerToDisplay === display.key) {
-                                        return ( 
-                                            <div className="react-block-list-container-header-text" style={{height: "16px", display: "flex", marginBottom: "16px", marginTop: "16px"}}>
-                                                <h3 style={{marginLeft: "5%", fontWeight: "bold", fontSize: "18px"}}>{row[display.key]}</h3>
-                                            </div>
-                                        );  
+                                    if (headersToDisplay.includes(display.key)) { 
+                                        if (headersCounter < headersToDisplay.length - 1) {
+                                            headerText += row[display.key] + " ";
+                                            headersCounter += 1;
```

Comment:
As far as I understand you are returning value ONLY for the last header to display. When using `map` function a value should be returned from every callback: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#when_not_to_use_map

Maybe you could iterate trough the headers outside of `return()` function, use `reduce` or `forEach` function to aggregate the string and display it only for the last header.

Reply:
yes you are right that this approach was bad. Here I commited https://github.com/OpenLMIS/openlmis-stockmanagement-ui/pull/24/commits/f4306c9bcf34f8fde8362e3081adb9630a7ca8f8 this change. Will be merged into this PR.